### PR TITLE
fixing termux backend for charging status and termux backend checker

### PR DESCRIPTION
**Pull request by xu3s**
It should fix battery status to show as always charging and it seems like placing termux backend checker (i don't know what it's called) below freebsd checker broke termux backend as there was exist sysctl, it fixed by just moving the checker upward

Ps. My English is bad

### CODE REVIEW

File: autoload/battery.vim [hunk 1]
```diff
@@ -93,10 +93,11 @@ function! s:get_available_backend() abort
     return 'powershell'
   elseif battery#backend#linux#is_available()
     return 'linux'
+	elseif battery#backend#termux#is_available()
```
Comment (lambdalisue) [inline autoload/battery.vim]: Please fix the indent level.

File: autoload/battery/backend/termux.vim [hunk 2]
```diff
@@ -19,7 +19,7 @@ endfunction
 
 function! s:on_exit(backend, buffer, exitval) abort
   let content = join(a:buffer, '')
-  let a:backend.is_charging = json_decode(content).status !=# 'NOT_CHARGING'
+  let a:backend.is_charging = json_decode(content).status !=# 'DISCHARGING'
```
Comment (lambdalisue) [inline autoload/battery/backend/termux.vim]: Why did you change the term?

File: autoload/battery/backend/termux.vim [hunk 3]
```diff
@@ -19,7 +19,7 @@ endfunction
 
 function! s:on_exit(backend, buffer, exitval) abort
   let content = join(a:buffer, '')
-  let a:backend.is_charging = json_decode(content).status !=# 'DISCHARGING'
+  let a:backend.is_charging = json_decode(content).status !=# 'NOT_CHARGING' || 'DISCHARGING'
```
Comment (lambdalisue) [inline autoload/battery/backend/termux.vim]: This fix doesn't make sense. What did you mean?

**Review by lambdalisue — CHANGES_REQUESTED** [review]

File: autoload/battery/backend/termux.vim (hunk 2 above)
Comment (xu3s) [reply autoload/battery/backend/termux.vim]: Because mine said its DISCHARGING instead of NOT_CHARGING

File: autoload/battery/backend/termux.vim (hunk 3 above)
Comment (xu3s) [reply autoload/battery/backend/termux.vim]: As i was scrolling here i see that the first person who PR this feature use NOT_CHARGING so i add the or statement

File: autoload/battery.vim (hunk 1 above)
Comment (xu3s) [reply autoload/battery.vim]: I didn't change the indent level tho, what do you mean?

File: autoload/battery/backend/termux.vim (hunk 3 above)
Comment (xu3s) [reply autoload/battery/backend/termux.vim]: It seems like my understanding on vim script is zero, can you help me? Or just reject this PR if you think its just a bother for you, I'm sorry for taking your time

I can't seem to understand how to check if one of those string is exist

File: autoload/battery.vim [hunk 4]
```diff
@@ -93,10 +93,10 @@ function! s:get_available_backend() abort
...
```
Comment (lambdalisue) [inline autoload/battery.vim]: You are using TAB here right? The correct indent here is TWO SPACES like:

```suggestion
  elseif battery#backend#termux#is_available()
```

File: autoload/battery/backend/termux.vim [hunk 5]
```diff
@@ -18,8 +18,11 @@ function! s:on_stdout(buffer, data) abort
 endfunction
 
 function! s:on_exit(backend, buffer, exitval) abort
+	let g:bat_status_list = ['NOT_CHARGING' , 'DISCHARGING']
   let content = join(a:buffer, '')
-  let a:backend.is_charging = json_decode(content).status !=# 'NOT_CHARGING'
+	if index(g:bat_status_list, json_decode(content).status) >= 0
+		let a:backend.is_charging = 0
+	endif
   let a:backend.value = json_decode(content).percentage + 0
```
Comment (lambdalisue) [inline autoload/battery/backend/termux.vim]: Ok. Rewrite it like the below. **Use TWO SPACES instead of TAB to indent**.

```vim
function! s:on_exit(backend, buffer, exitval) abort
  let obj = json_decode(join(a:buffer, '{}'))
  let status = get(obj, 'status', '')
  let a:backend.is_charging = !(status ==# 'NOT_CHARGING' || status ==# 'DISCHARGING')
  let a:backend.value = get(obj, 'percentage', 0) + 0
endfunction
```

I didn't check if the above really work so please confirm the behavior by yourselves.

**Review by lambdalisue — CHANGES_REQUESTED** [review]

File: autoload/battery.vim (hunk 4 above)
Comment (xu3s) [reply autoload/battery.vim]: Ok

File: autoload/battery/backend/termux.vim (hunk 5 above)
Comment (xu3s) [reply autoload/battery/backend/termux.vim]: This work, i have push it, thank you

**Review by lambdalisue — APPROVED** [review]

**Comment by lambdalisue** [comment]
Thanks for the contribution 👍
